internal/internal_workflow_client.go

```diff
@@ -367,6 +367,12 @@ func (wc *WorkflowClient) SignalWithStartWorkflow(ctx context.Context, workflowI
 		return nil, err
 	}
 
+	// Due to the ambiguous way to provide workflow IDs, if options contains an
+	// ID, it must match the parameter
+	if options.ID != "" && options.ID != workflowID {
+		return nil, fmt.Errorf("workflow ID from options not used, must be unset or match workflow ID parameter")
```

Comment:
This is technically a breaking change.
I think it's necessary but we should document this in the release notes and mark it as breaking.

Reply:
It's only a breaking change if you consider extra validation of (unbeknownst) improper param setting as a breaking change. It went from being an invalid param do the wrong thing to an invalid param error.

Reply:
I'm aware, but this still could result in a change of behavior in some rare cases.

Reply:
:+1: Ok, I will make sure to document in the release notes